### PR TITLE
Drucker-Prager wellbore with cohesion hardening

### PR DESCRIPTION
**Pull request by sytuannguyen**
This PR provides xmls, post-processing scripts, analytical solutions and docs for Drucker-Prager wellbore problem with cohesion hardening. 

Smoke example is added to integratedTests https://github.com/GEOS-DEV/integratedTests/pull/9

### CODE REVIEW

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+| :math:`\sigma_h` | Horizontal Stress       | [MPa]            | -11.25        |
++------------------+-------------------------+------------------+---------------+
+| :math:`\sigma_v` | Vertical Stress         | [MPa]            | -15.0         |
++------------------+-------------------------+------------------+---------------+
+| :math:`a_0`      | Initial Well Radius     | [m]              | 0.1           |
++------------------+-------------------------+------------------+---------------+
+| :math:`p_w`      | Mud Pressure            | [MPa]            | -2.0          |
++------------------+-------------------------+------------------+---------------+
+
+The validations of GEOS results against analyticla results is shown in the figure below:
```
Comment (herve-gross) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst:69]:
```suggestion
The validation of GEOS results against analytical results is shown in the figure below:
```

**Review by herve-gross — APPROVED** [review]
Looks great! Thank you.

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst (hunk 1 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst]: Done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+
+#########################################################
+Drucker-Prager Model with Hardening for Wellbore Problems
+#########################################################
+
+
+**Context**
+
+This is an alternative to the example :ref:`ExampleEDPWellbore` where the Drucker-Prager constitutive with cohesion hardening (see :ref:`DruckerPrager`) is considered. Analytical solutions to this problem are not provided from open literature but they can be derived following the guide of (Chen and Abousleiman 2017)  <https://www.sciencedirect.com/science/article/pii/S1365160917301090>`__. Details of those solutions are given in Python scripts associated to this example.
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst:11]:
```suggestion
This is an alternative to the tutorial example :ref:`ExampleEDPWellbore`, and the Drucker-Prager constitutive with cohesion hardening (see :ref:`DruckerPrager`) is hereby considered. Analytical solutions to this problem are not provided from literature work, however they can be derived following `(Chen and Abousleiman 2017)  <https://www.sciencedirect.com/science/article/pii/S1365160917301090>`__. Details of those solutions are given in Python scripts associated to this example.
```

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  inputFiles/solidMechanics/DruckerPragerWellbore_benchmark.xml
+
+The related integrated test is
+
+.. code-block:: console
+
+  inputFiles/solidMechanics/DruckerPragerWellbore_smoke.xml
+
+
+The Drucker-Prager properties are specified in the ``Constitutive`` section:
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst:33]:
```suggestion
The Drucker-Prager material properties are specified in the ``Constitutive`` section: 
```

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+The Drucker-Prager properties are specified in the ``Constitutive`` section: 
+
+.. literalinclude:: ../../../../../../../inputFiles/solidMechanics/DruckerPragerWellbore_base.xml
+    :language: xml
+    :start-after: <!-- SPHINX_DP_WELLBORE_MATERIAL -->
+    :end-before: <!-- SPHINX_DP_WELLBORE_MATERIAL_END -->
+
+
+Here, ``rock`` is designated as the material in the computational domain. Drucker Prager model ``DruckerPrager`` is used to simulate the elastoplastic behavior of ``rock``. The material parameters, ``defaultFrictionAngle``, ``defaultDilationAngle`` and ``defaultCohesion`` denote the friction angle, the dilation angle, and the cohesion, respectively. In this example, cohesion is increased by a linear hardening law that is defined by the parameter ``defaultHardeningRate``. The constitutive parameters such as the density, the bulk modulus, and the shear modulus are specified in the International System of Units.
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst:41]:
```suggestion
Here, ``rock`` is designated as the material in the computational domain. Drucker Prager model ``DruckerPrager`` is used to simulate the elastoplastic behavior of ``rock``. The material parameters, ``defaultFrictionAngle``, ``defaultDilationAngle`` and ``defaultCohesion`` denote the friction angle, the dilation angle, and the cohesion, respectively. In this example, the hardening of cohesion is described by a linear hardening law, which is governed by the parameter ``defaultHardeningRate``. The constitutive parameters such as the density, the bulk modulus, and the shear modulus are specified in the International System of Units.
```

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/elastoPlasticWellboreAnalyticalSolutions.py [hunk 5]
```diff
@@ -8,18 +8,18 @@ def invariants(sr,s0,sz):
...
-def compute_a(sr,s0,sz,sigma,dFdp, dFdq):
+def compute_a(sr,s0,sz,sigma,dGdp, dGdq):
     p,q = invariants(sr,s0,sz)
     dpdsigma = 1.0/3.0
     dqdsigma = 3.0*(sigma-p)/2.0/q
-    return dFdp*dpdsigma + dFdq*dqdsigma
+    return dGdp*dpdsigma + dGdq*dqdsigma
 
-def compute_bij(sr, s0, sz, dFdp, dFdq, E, nu, dFdpc,dpcdepsVp,dGdp):
+def compute_bij(sr, s0, sz, dGdp, dGdq, E, nu, dFdpc,dpcdepsVp,dGdp):
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/elastoPlasticWellboreAnalyticalSolutions.py:17]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/elastoPlasticWellboreAnalyticalSolutions.py [hunk 6]
```diff
@@ -42,11 +42,11 @@ def compute_bij(sr, s0, sz, dFdp, dFdq, E, nu, dFdpc,dpcdepsVp,dGdp):
     return b11,b12,b13,b22,b23,b33,b21,b31,b32,delta
     
 
-def solution_plastic(sr, s0, sz, epsV, xd, dx, dFdp, dFdq, E, nu, dFdpc,dpcdepsVp,dGdp):
-    b11,b12,b13,b22,b23,b33,b21,b31,b32,delta = compute_bij(sr, s0, sz, dFdp, dFdq, E, nu, dFdpc,dpcdepsVp,dGdp)
+def solution_plastic(sr, s0, sz, epsV, xd, dx, dGdp, dGdq, E, nu, dFdpc,dpcdepsVp,dGdp):
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/elastoPlasticWellboreAnalyticalSolutions.py:45]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst (hunk 2 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst:11]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst (hunk 3 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst:33]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst (hunk 4 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst:41]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/dpWellbore_plot.py [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+	plt.yticks(np.linspace(0.0,20e6,11))
+	plt.xscale('log', subs=range(2,10))
+	plt.legend()
+
+	# Plot GEOS results for elasto-plastic material versus analytical results for elastic material
+	r_elas_assumed,srr_elas_assumed,stt_elas_assumed,szz_elas_assumed = epwAnal.solution_elastic(sh,sv,1.0,pw)
+	
+	plt.subplot(2,2,2)
+	plt.plot( rCoord/rCoord[0],-stress_rr,'r+',label=r'$\sigma_{rr}$: GEOS')
+	plt.plot(r_elas_assumed,srr_elas_assumed, 'r', label=r'$\sigma_{rr}$: Analytic Elastic')
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/dpWellbore_plot.py:186]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/dpWellbore_plot.py [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+
+	plt.plot( rCoord/rCoord[0],-stress_tt,'b+',label=r'$\sigma_{\theta\theta}$: GEOS')
+	plt.plot(r_elas_assumed,stt_elas_assumed, 'b', label=r'$\sigma_{\theta\theta}$: Analytic Elastic')
+
+	plt.plot( rCoord/rCoord[0],-stress_zz,'g+',label=r'$\sigma_{zz}$: GEOS')
+	plt.plot(r_elas_assumed,szz_elas_assumed, 'g', label=r'$\sigma_{zz}$: Analytic Elastic')
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/dpWellbore_plot.py:192]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/dpWellbore_plot.py [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+	# Plot the stress path on the p-q plan when the well surface pressure decrease from the in-situ condition
+	defaultFrictionAngle *= np.pi/180.0 # converted to rad
+	param_b = dpAnal.compute_param_b(defaultFrictionAngle)
+	param_a = defaultCohesion * param_b / np.tan(defaultFrictionAngle)
+	p_yieldSurface = np.arange(0,15e6,1)
+	q_yieldSurface_i = p_yieldSurface*param_b + param_a
+
+	plt.subplot(2,2,3)
+	plt.plot(-p,q,'ko',label='GEOS')
+	plt.plot(list_p_wellsurface_analytic,list_q_wellsurface_analytic,'b',label='Analytic')
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/dpWellbore_plot.py:211]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/dpWellbore_plot.py [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+	#plt.xlim(0,15e6)
+	#plt.ylim(0,10e6)
+	plt.legend()
+
+	# Plot the wellbore deformation a0/a versus wellbore surface pressure pw
+	pw = stress_xx_wellboresurface
+
+	plt.subplot(2,2,4)
+	plt.plot(a0/a, -pw,'ko',label='GEOS')
+	plt.plot(list_a0_a_ratio,list_pw_analytic,'b',label='Analytic')
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/dpWellbore_plot.py:225]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/dpWellbore_plot.py [hunk 11]
```diff
@@ -0,0 +1,236 @@
...
+		list_p_wellsurface_analytic.append(p_wellsurface_analytic)
+		list_q_wellsurface_analytic.append(q_wellsurface_analytic)
+	
+	# Plots	
+	plt.figure(figsize=(15,10))
+
+	# Plot GEOS results versus analytical results for elasto-plastic material
+	plt.subplot(2,2,1)
+	plt.plot( rCoord/rCoord[0],-stress_rr,'r+',label=r'$\sigma_{rr}$: GEOS')
+	plt.plot(r_ep_analytic,srr_ep_analytic, 'r', label=r'$\sigma_{rr}$: Analytic Plastic')
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/dpWellbore_plot.py:160]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/dpWellbore_plot.py [hunk 12]
```diff
@@ -0,0 +1,236 @@
...
+	plt.plot(r_elas_analytic,srr_elas_analytic, 'r')
+
+	plt.plot( rCoord/rCoord[0],-stress_tt,'b+',label=r'$\sigma_{\theta\theta}$: GEOS')
+	plt.plot(r_ep_analytic,stt_ep_analytic, 'b', label=r'$\sigma_{\theta\theta}$: Analytic Plastic')
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/dpWellbore_plot.py:164]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/dpWellbore_plot.py [hunk 13]
```diff
@@ -0,0 +1,236 @@
...
+	plt.plot(r_elas_analytic,stt_elas_analytic, 'b')
+
+	plt.plot( rCoord/rCoord[0],-stress_zz,'g+',label=r'$\sigma_{zz}$: GEOS')
+	plt.plot(r_ep_analytic,szz_ep_analytic, 'g', label=r'$\sigma_{zz}$: Analytic Plastic')
```
[the remark this answers is not in the document]
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/dpWellbore_plot.py:168]: done

**Review by jhuang2601 — APPROVED** [review]

**Comment by francoishamon** [comment]
I am working on the integrated tests rebaseline for this PR

**Pull request merged**